src/pip/_internal/cli/cmdoptions.py

```diff
@@ -480,24 +481,49 @@ def only_binary():
 
 # This was made a separate function for unit-testing purposes.
 def _convert_python_version(value):
-    # type: (str) -> Tuple[int, ...]
+    # type: (str) -> Tuple[Tuple[int, ...], Optional[str]]
     """
-    Convert a string like "3" or "34" into a tuple of ints.
+    Convert a version string like "3", "37", or "3.7.3" into a tuple of ints.
+
+    :return: A 2-tuple (version_info, error_msg), where `error_msg` is
```

Comment:
This seems strange, why not `raise` in such case ?

Reply:
It is raised by the caller using `raise_option_error()` and the message returned by this function. It's easier to test this way because `raise_option_error()` requires both an `Option` and `OptionParser` object, and it seems like it would be an unnecessary complication to raise a `ValueError` just so the caller can catch it, get the message from the exception's argument, and then re-raise using `raise_option_error()`. (This function is used in only one place and was separated out only to make testing the `--python-version` argument handling inside `_handle_python_version()` easier.)

Reply:
> It's easier to test this way

I suspected so but this `return (result, error)` feels strange in a python program (and looks more like golang)...
Raising and Exception would be expected (and could be tested with an assertRaise), so I guess it is a matter of taste :)